core/templates/pages/admin-page/roles-tab/admin-roles-tab.component.spec.ts

```diff
@@ -0,0 +1,806 @@
+
+      const viewContributionReviewersAction: ViewContributionReviewersAction = {
+        category: 'category',
+        filterCriterion: 'username',
+        isValid: () => true,
+        languageCode: 'en',
+        username: 'user1'
+      };
+
+      expect(component.contributionReviewersDataFetched).toBe(false);
+
+      component.submitViewContributorUsersForm(viewContributionReviewersAction);
+      tick();
+
+      expect(adminBackendServiceSpy).not.toHaveBeenCalled();
+      expect(component.contributionReviewersDataFetched).toBe(false);
+    }));
+  });
+
+  describe('on clicking add rights button ', () => {
+    it('should successfully update the rights of the user', fakeAsync(() => {
+      let adminBackendServiceSpy = spyOn(
+        adminBackendApiService, 'addContributionReviewerAsync')
+        .and.returnValue(Promise.resolve(null));
+
+      const addContributionRightsAction: AddContributionRightsAction = {
+        category: 'translation',
+        isValid: () => true,
+        languageCode: 'en',
+        username: 'user1'
+      };
+
+      component.submitAddContributionRightsForm(addContributionRightsAction);
+      tick();
+
+      expect(adminBackendServiceSpy).toHaveBeenCalled();
+      expect(statusMessageSpy).toHaveBeenCalledWith(
+        'Successfully added "user1" as translation reviewer.');
+    }));
+
+    it('should not send request to backend if a task ' +
+      'is still running in the que', fakeAsync(() => {
+      // Setting task running to be true.
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(true);
+
+      let adminBackendServiceSpy = spyOn(
+        adminBackendApiService, 'addContributionReviewerAsync')
+        .and.returnValue(Promise.resolve(null));
+
+      const addContributionRightsAction: AddContributionRightsAction = {
+        category: 'translation',
+        isValid: () => true,
+        languageCode: 'en',
+        username: 'user1'
+      };
+
+      component.submitAddContributionRightsForm(addContributionRightsAction);
+      tick();
+
+      expect(adminBackendServiceSpy).not.toHaveBeenCalled();
+    }));
+  });
+
+  describe('on clicking remove rights button ', () => {
+    it('should successfully remove the rights of the user', fakeAsync(() => {
+      let adminBackendServiceSpy = spyOn(
+        adminBackendApiService, 'removeContributionReviewerAsync')
+        .and.returnValue(Promise.resolve(null));
+
+      const removeContributionRightsAction: RemoveContributionRightsAction = {
+        category: 'translation',
+        isValid: () => true,
+        languageCode: 'en',
+        method: 'all',
+        username: 'user1'
+      };
+
+      component.submitRemoveContributionRightsForm(
+        removeContributionRightsAction);
+      tick();
+
+      expect(adminBackendServiceSpy).toHaveBeenCalled();
+      expect(statusMessageSpy).toHaveBeenCalledWith(
+        'Success.');
+    }));
+
+    it('should not send request to backend if a task ' +
```

Comment:
This test description seems to be repeated throughout the file -- maybe modify it to describe the type of request, so that the descriptions are distinguishable

Reply:
please note that we have different describe blocks for each case.